src/modules/land_detector/RoverLandDetector.cpp

```diff
@@ -44,14 +44,31 @@
 namespace land_detector
 {
 
+
 bool RoverLandDetector::_get_ground_contact_state()
 {
 	return true;
 }
 
 bool RoverLandDetector::_get_landed_state()
 {
-	return !_actuator_armed.armed;
+
+// I need to do the uORB subscription here, can't do it in the .hpp file (SIGSEGV if I do... but no idea why)
+uORB::Subscription vehicle_status_sub{ORB_ID(vehicle_status)};
+vehicle_status_s vehicle_status{};		/**< vehicle status */
+
+vehicle_status_sub.update(&vehicle_status);
+
+if(vehicle_status.nav_state == vehicle_status_s::NAVIGATION_STATE_AUTO_LAND)
+{
+	return true; // If Landing has been requested then say we have landed
+}
+else
+{
+	return !_actuator_armed.armed;  // If we are armed we are not landed
```

Comment:
The system being disarmed should probably be checked first to override.

Reply:
Hello dagar, I don't really follow what you mean. This part of the code is unchanged.

Reply:
I mean you can check `_actuator_armed.armed` before updating vehicle_status and checking the nav_state.

Reply:
Ah, I think my comment in the code might be confusing things. The system works perfectly as it is currently.

The original part of the code (which is now part of the else) is what didn't work properly 
```
 else {
		return !_actuator_armed.armed;  // If we are armed we are not landed
		// This doesn't work very well because once it's armed it will stay armed
		// so when requesting land it never detects it has landed.
```
I'll remove those comments, as they are confusing

Reply:
Done